app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb

```diff
@@ -5,7 +5,8 @@ module ResourceRestrictions
 
     ExtractResourceRestrictions = CommandClass.new(
       dependencies: {
-        resource_restrictions_class: ResourceRestrictions::ResourceRestrictions,
+        resource_restrictions_class: Authentication::ResourceRestrictions::ResourceRestrictions,
+        get_restrictions_class: Authentication::ResourceRestrictions::GetRestrictionsFromAnnotation,
```

Comment:
You don't need the class, you can hold the instance as a dependency:
```suggestion
        get_restriction_from_annotation: Authentication::ResourceRestrictions::GetRestrictionsFromAnnotation.new,
```

Reply:
Done